File: 1548-check-if-all-1s-are-at-least-length-k-places-away/check-if-all-1s-are-at-least-length-k-places-away.java
class Solution {
    public boolean kLengthApart(int[] nums, int k) {
        int n = nums.length;

        for (int i = 0, prev = -10000; i < n; i++) {
            if (nums[i] == 1) {
                if (i - prev - 1 < k) {
                    return false;
                }
                prev = i;
            }
        }

        return true;
    }
}

File: 1548-check-if-all-1s-are-at-least-length-k-places-away/check-if-all-1s-are-at-least-length-k-places-away.py
class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        n = len(nums)
        
        prev = -100000
        for i in range(n):
            if 1 == nums[i]:
                if i - prev - 1 < k:
                    return False
                else:
                    prev = i

        return True
